### PR TITLE
Move loop creation into `BaseWorker` class

### PR DESCRIPTION
**Pull request by desertaxle**
This PR refactors the `BaseWorker` to own starting polling loops instead of the CLI. This is in anticipation of removing cancellation from workers when removing the `PREFECT_EXPERIMENTAL_ENABLE_ENHANCED_CANCELLATION` setting. The `ProcessWorker` will need to keep its cancellation loop (because it doesn't start a `Runner` for flow runs), so this PR adds a `.start` method to handle loop creation, which the `ProcessWorker` will be able to implement to maintain a cancellation loop.

Related to https://github.com/PrefectHQ/prefect/issues/14208 and https://github.com/PrefectHQ/prefect/issues/14209

### Checklist

- [x] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
- [x] This pull request references any related issue by including "closes `<link to issue>`"
  
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] If this pull request adds new functionality, it includes unit tests that cover the changes
- [ ] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [ ] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

File: src/prefect/workers/base.py [hunk 1]
```diff
@@ -511,6 +519,87 @@ def get_flow_run_logger(self, flow_run: "FlowRun") -> PrefectLogAdapter:
...
+                    webserver_thread = threading.Thread(
+                        name="healthcheck-server-thread",
+                        target=partial(
+                            start_healthcheck_server,
+                            worker=self,
+                            query_interval_seconds=PREFECT_WORKER_QUERY_SECONDS.value(),
+                        ),
+                        daemon=True,
+                    )
+                    webserver_thread.start()
```
Comment (chrisguidry) [inline src/prefect/workers/base.py:598]: I think we will want to kill this thread proactively in a `finally`, right?  If not, the thread won't exit until the process exits, which could lead to a memory leak if you were doing something interesting like starting & stopping multiple workers, etc.

**Review by chrisguidry — APPROVED** [review]
Very clean!  I do think we want to explicitly clean up that health check web server thread, though

File: src/prefect/workers/base.py (hunk 1 above)
Comment (desertaxle) [reply src/prefect/workers/base.py]: Good call! I added handling for that in 911b2dc9cbc638d0b7c6cc970309f2f4e93f7e6f

Comment (chrisguidry) [reply src/prefect/workers/base.py]: Rad, looks great!

**Pull request merged**